MarsRover.Tests/ControllerShould.cs

```diff
@@ -16,26 +16,89 @@ public class ControllerShould
             test for 
         */
         [Fact]
-        public void Setup_ReturnsRoverLocationAndDirection_GivenValidMapAndRoverData()
+        public void Setup_ReturnsMapOutput_GivenValidMapAndRoverData()
         {
             var mockInput = new Mock<IInput>();
-            mockInput.SetupSequence(_ => _.Read())
+            mockInput.SetupSequence(_ => _.ReadLine())
                 .Returns("TestFiles/validFile1.txt")
                 .Returns("N")
                 .Returns("2,0");
             var output = new StubOutput();
-            var mockMapInput = new Mock<IMapInput>();
-            mockMapInput.Setup(_ => _.Read("TestFiles/validFile1.txt"))
-                .Returns("ONNN\nNNNN\nNNNN");
-            mockMapInput.Setup(_ => _.FileExists("TestFiles/validFile1.txt"))
-                .Returns(true);
-            var controller = new Controller(mockInput.Object, output, mockMapInput.Object); 
-
-            controller.Setup(); 
-
-            controller.Rover.Direction.Should().Be(Direction.North);
-            controller.Rover.Location.X.Should().Be(2);
-            controller.Rover.Location.Y.Should().Be(0);
+            var fileMapInput = new FileMapInput();
+            fileMapInput.Read("TestFiles/validFile1.txt");
+            fileMapInput.FileExists("TestFiles/validFile1.txt");

```

Comment:
if you are using the actual file map input then you don't need to call read or file exists outside of the controller like this as you don't need to do any set up for it the way that you need to for mocks. These methods are already invoked in your InitialiseMap method

Reply:
Think of these calls as "implementation details", for which you test implicitly through behaviour rather than explicitly through making the direct calls in the tests